Expand and clarify the PiTR documentation. (#513)
vitessio/website: content/en/docs/reference/features/recovery.md

### Use cases
- Accidental deletion of data, e.g. dropping a table by mistake, running an UPDATE or DELETE with an incorrect WHERE clause, etc.
- Corruption of data due to application bugs.
- Corruption of data due to MySQL bugs or underlying hardware (e.g. storage) problems.

### Preconditions
- There should be a Vitess backup taken before the desired point in time.
- There should be continuous binlogs available from the backup time to the desired point in time.
- This feature is tested using [Ripple](https://github.com/google/mysql-ripple) as the binlog server.  However, it should be possible to use a MySQL instance as source for the binlogs as well.

### Example usage

To use this feature, you need a usable backup of Vitess data and continuous binlogs.

Here is how you can create a backup.

```sh
$ vtctlclient -server <vtctld_host>:<vtctld_port> Backup zone1-101
```

Here `zone1-101` is the tablet alias of a replica tablet in the shard that you
want to back up.  Note that you can also use `vtctlclient BackupShard` to just
specify a keyspace and shard, and have Vitess choose the tablet to run the

> backup on for you, instead of having to specify the tablet alias explicitly.

Reviewer: ```suggestion
backup for you, instead of having to specify the tablet alias explicitly.
```

To maintain continuous binlogs, you need to have a binlog server pointing to
the master (or a replica, assuming that the replica is also maintaining its
own binlogs, which is the default Vitess configuration). You can use
[Ripple](https://github.com/google/mysql-ripple) as a binlog server, although

> there are other options;  and you could use an existing MySQL server as well.

Reviewer: ```suggestion
there are other options; and you could use an existing MySQL server as well.
```

```

 Here:
 - `originalks` is the base keyspace, i.e. the keyspace we took a backup of,
 and are trying to recover.
 - `snapshot_time` is the timestamp of the point in time to we want to recover
 to. Note the use of the `Z` in the timestamp, indicating it is expressed
 in UTC.
 - `restoreks` is the name of recovery keyspace, i.e. the keyspace to which
 we are restoring our backup.

 Next, you can launch the vttablet, which as part of vttablet's normal
 initialization procedure will look for a backup to restore. It will
 detect the meta-information you added on the keyspace topology node
 when creating the keyspace above.  It will then use that information
 to restore the last backup earlier than the timestamp provided for the
 specific shard the vttablet is in.

 Here are the command line arguments vttablet uses in this
 process.  You may already be using some of these as part of your
 normal vttablet initialization parameters (e.g. if you are using the
 Vitess K8s operator):
 - `-init_keyspace restoreks` - here `restoreks` is the recovery keyspace
 name which we created earlier

>  - `-init_db_name_override vt_originalks` - here `vt_originalks` is the

Reviewer: One thing to add here is that if the tablets for originalks set this flag to a specific value, then the same value needs to be set for the recovery tablet.

 If you use this option, you will also need to specify:
   - `-file_backup_storage_root` - with a path pointing to your backup
 storage location.
 - `-backup_storage_implementation s3` - for backing up to S3. If you
 use this option, you may need additional flags like:
   - `-s3_backup_aws_region`
   - `-s3_backup_storage_bucket`
   - `-s3_backup_storage_root`
 - There are more `-backup_storage_implementation` options like `gcs` and
  others.

You will also probably want to use other flags for backup and restore like:
 - `-backup_engine_implementation xtrabackup` - Use Percona Xtrabackup to
 take online backups. Without this flag, the mysql instance on the replica
 being backed up will be shut down during the backup.
 - `-backup_storage_compress true` - gzip compress the backup (default is
 true).
You need to be consistent in your use of these flags for backup and restore.

Once the restore of the last backup earlier than the `snapshot_time` timestamp
is completed, the vttablet proceeds to use the `binlog_*` parameters to
connect to the binlog server and then apply all binlog events from the time
of the backup until the timestamp provided.

> Since the last backup for each shard making up the keyspace are taken at

Reviewer: ```suggestion
Since the last backup for each shard making up the keyspace could be taken at
```

different points in time, the amount of time that it takes to apply these events
may differ between restores of different shards in the keyspace.

Note that to restore to the specified `snapshot_time` timestamp, vttablet needs
to find the GTID corresponding to the last event before this timestamp from
the binlog server. This is an expensive operation and may take some time. By
default the timeout for this operation is one minute (1m). This can be changed
by setting the vttablet `-pitr_gtid_lookup_timeout` flag.

VTGate will automatically exclude tablets belonging to snapshot keyspaces from

> query routing unless they are specifically addressed using `USE originalks`

Reviewer: this should be `USE restoreks`
